modules/@apostrophecms/schema/lib/addFieldTypes.js

```diff
@@ -105,26 +105,37 @@ module.exports = (self) => {
         throw self.apos.error('required');
       }
 
-      if (
-        field.pattern &&
-        field.pattern instanceof RegExp &&
-        !field.pattern.test(destination[field.name]
-        )) {
-        throw self.apos.error('invalid');
+      if (field.pattern) {
+        const regex = new RegExp(field.pattern);
+
+        if (!regex.test(destination[field.name])) {
+          throw self.apos.error('invalid');
+        }
       }
     },
-    index: function (value, field, texts) {
+    index(value, field, texts) {
       const silent = field.silent === undefined ? true : field.silent;
       texts.push({
         weight: field.weight || 15,
         text: value,
         silent: silent
       });
     },
-    isEmpty: function (field, value) {
+    isEmpty(field, value) {
       return !value.length;
     },
-    addQueryBuilder: function (field, query) {
+    validate(field, options, warn, fail) {
+      if (!field.pattern) {
+        return;
+      }
+
+      if (!(field.pattern instanceof RegExp)) {
+        fail('The pattern property must be a RegExp');
```

Comment:
If it is a string, I think we have to accept that because everything else in schemas currently is 100% json-friendly and we will want to be able to build schemas with JSON-based tools in the future. However, you can bounce that string through the RegExp constructor and complain if it throws an exception.

Reply:
It's about supporting both.